pkg/storage/postgresql/targets.go

```diff
@@ -171,6 +173,75 @@ func (db DB) getTargetTags(id string) ([]string, error) {
 	return tags, err
 }
 
+// CreateTarget creates the given target. If the target contains
+// tags associated, these are created as part of the same transaction.
+// If target or associated tags already exist, no error is returned.
+func (db DB) CreateTarget(t model.Target) (model.Target, error) {
+	tx, err := db.DB.BeginTxx(context.Background(), nil)
+	if err != nil {
+		return model.Target{}, err
+	}
+
+	var created model.Target
+	var tags []uint8
+
+	query := `
+	WITH target_exists AS (
+		SELECT t.*,
+		(
+			SELECT ARRAY_AGG(tt.tag)
+			FROM target_tags tt
+			WHERE t.id = tt.target_id
+		) tags
+		FROM targets t
+		WHERE t.identifier = $1
+	),
+	creation_query AS (
+		INSERT INTO targets (identifier)
+		SELECT $1
+		WHERE NOT EXISTS (SELECT 1 FROM target_exists)
+		RETURNING *, ARRAY[]::text[] AS tags
+	)
+	SELECT * FROM target_exists
+    UNION ALL
+	SELECT * FROM creation_query`
+	err = tx.QueryRow(query, t.Identifier).Scan(
+		&created.ID, &created.Identifier, &tags,
+	)
+	if err != nil {
+		tx.Rollback()
+		return model.Target{}, err
+	}
+
+	created.Tags = append(t.Tags, parseStringArray(tags)...)
```

Comment:
Imagine that we are trying to insert a new target. The `creation_query` will return `ARRAY[]::text[]` as `tags`.
In this scenario, if the user has provided `tags`, they will be ignored, right?

Is this intended?